Add tale

notcharlie2026 opened the pull request.

notcharlie2026 committed:
Create Coralmech.java

Ben-Vs-Bread committed:
added coral mech and algae mech

notcharlie2026 committed:
initial generated, building no cancoder

notcharlie2026 committed:
added cancoder

Frank0624789 committed:
Generated elevator code based on leader

doesnt build yet bc closedloopramprate variable

Frank0624789 committed:
fixed it up

Frank0624789 committed:
implementation of can

Frank0624789 committed:
added limit switch stuff (still js for leader)

notcharlie2026 committed:
Update DragonElevator.java

notcharlie2026 committed:
Merge pull request #22 from Team302/ElevatorMechanism😭

Elevator mechanism😭 FOR NOW

notcharlie2026 committed:
made leader and follower work maybe

notcharlie2026 committed:
fixed naming

notcharlie2026 committed:
Create TaleTeller.java

notcharlie2026 committed:
fixed build error

notcharlie2026 committed:
base state and mag offsets

notcharlie2026 committed:
Update TaleTeller.java

notcharlie2026 committed:
Delete Algaemech.java

notcharlie2026 committed:
Create AlgaeMech.java

notcharlie2026 committed:
fix build

notcharlie2026 committed:
Update TaleTeller.java

notcharlie2026 committed:
Delete Coralmech.java

notcharlie2026 committed:
Create CoralMech.java

notcharlie2026 committed:
made sim work kinda

notcharlie2026 committed:
removed tale teller

notcharlie2026 committed:
Update RobotContainer.java

notcharlie2026 committed:
fixed can stuff, working on command trigger

notcharlie2026 committed:
fixed it 👍

Frank0624789 committed:
updated setcontrol stuff algae mech😋😋😋

notcharlie2026 committed:
Merge branch 'main' into add-Tale-cmd-groups

notcharlie2026 committed:
Merge branch 'add-TaleTeller' of https://github.com/Team302/2025ReefscapeJavaCommand into add-Tale-cmd-groups

notcharlie2026 committed:
Revert "Merge branch 'add-TaleTeller' of https://github.com/Team302/2025ReefscapeJavaCommand into add-Tale-cmd-groups"

This reverts commit cb293bcbbe75ecd151049772171e186288309467, reversing
changes made to 8027223d72e6db0360e38fa9604ff6e304cef406.

Commit:
Google Java Format

notcharlie2026 committed:
Revert "Merge branch 'main' into add-Tale-cmd-groups"

This reverts commit 8027223d72e6db0360e38fa9604ff6e304cef406, reversing
changes made to 13b40a8b954fbb03890015b406a4189639215f53.

notcharlie2026 committed:
Merge branch 'add-TaleTeller' of https://github.com/Team302/2025ReefscapeJavaCommand into add-Tale-cmd-groups

notcharlie2026 committed:
Update RobotContainer.java

reslip1 commented on src/main/java/frc/robot/subsystems/drivetrain/CommandSwerveDrivetrain.java:
Why would we put a command under a system and why is drivetrain no longer a system?

notcharlie2026 replied:
@reslip1 we're using the convention where we are returning commands from the class, as opposed to making a whole new class just for commands. This is how the class was written and is direct copied from the ctre docs.

notcharlie2026 committed:
Merge branch 'main' into add-Tale-cmd-groups

notcharlie2026 committed:
re added the formatter and fixed build errors in robot container

Commit:
Google Java Format

notcharlie2026 committed:
Refactor subsystems to use unit-safe types

Replaced primitive double constants with unit-safe types (Voltage, Current, Angle, Distance, etc.) in AlgaeMech, ArmMech, CoralMech, and ElevatorMech. Updated all relevant configuration and calculation code to use these types, improving type safety and clarity for physical quantities throughout the subsystems. (copilot)

notcharlie2026 commented:
@witcpalek I think this is ready to be merged, and a repo created off of this main in the system core project

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+  // Motor controller
+  private final TalonFXS motor;
+  private final PositionVoltage positionRequest;
+  private final VelocityVoltage velocityRequest;
```
Should all of these be class wide variables?

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+  /** Creates a new Arm Subsystem. */
+  public CoralMech() {
+    // Initialize motor controller
+    motor = new TalonFXS(canID, "canivore");
```
It would be nice to have canbusname as a variable that is set to "canivore" near where canID gets set.

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+   * @return Position in Rotations
+   */
+  @Logged(name = "Position/Rotations")
+  public double getPosition() {
```
return a units-based value

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+   * @return Velocity in rotations per second
+   */
+  @Logged(name = "Velocity")
+  public double getVelocity() {
```
return a units-based value

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+   * @return Applied voltage
+   */
+  @Logged(name = "Voltage")
+  public double getVoltage() {
```
return a units-based value

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+   * @return Motor current in amps
+   */
+  @Logged(name = "Current")
+  public double getCurrent() {
```
return a units-based value

witcpalek commented on src/main/java/frc/robot/subsystems/Tale/CoralMech.java:
```diff
@@ -0,0 +1,239 @@
+   *
+   * @param voltage The voltage to apply
+   */
+  public void setVoltage(double voltage) {
```
use units-based parameters

witcpalek reviewed (changes requested):
@notcharlie2026 
It looks like there are a bunch of extra methods that are not being used.  We should only implement methods that are used unless there is another PR coming that is going to call these methods.   Extra methods just make files bigger and since the are not tested, if we ever want/need them, we assume they work, but they may not because they were never tested.

notcharlie2026 committed:
Disable desktop support and fix controller comments

Set includeDesktopSupport to false in build.gradle to disable desktop support. Updated comments in RobotContainer.java to correct controller class names and fix a typo in the documentation.

notcharlie2026 committed:
Update AlgaeMech.java

notcharlie2026 committed:
Update AlgaeMech.java

notcharlie2026 commented on src/main/java/frc/robot/subsystems/Tale/ArmMech.java:
```diff
@@ -0,0 +1,340 @@
+  // Motor controller
+  private final TalonFX m_motor;
+  private final CANcoder m_canCoder;
+  private final PositionVoltage m_positionRequest;
```
@witcpalek control requests have been made local, but the rest is being used by multiple functions.

Ben-Vs-Bread committed:
not working yet

Commit:
Google Java Format

notcharlie2026 committed:
Add TODO for joystick input negation and clean up AlgaeMech

Added a TODO in RobotContainer to address universal negation of joystick inputs by potentially wrapping CommandXboxController. Also removed an unused comment about gear ratio in AlgaeMech for code clarity.

notcharlie2026 committed:
Clean up comments and enable simulation input

Removed commented-out example code from RobotContainer and added a TODO about joystick input negation. In AlgaeMech, uncommented the line setting input voltage to the simulation to enable more accurate simulation behavior.

Ben-Vs-Bread committed:
Merge branch 'tale-development-' into add-TaleTeller

Ben-Vs-Bread committed:
made the code build

Commit:
Google Java Format

notcharlie2026 committed:
Refactor Climber to use direct motor signals

Removed cached StatusSignal fields and now access TalonFX motor signals directly for position, velocity, voltage, current, and temperature. Simplifies code, improves clarity, and ensures up-to-date values are used for telemetry and control.

notcharlie2026 committed:
Refactor AlgaeMech and CoralMech sensor access

Removed direct StatusSignal usage and simulation code from AlgaeMech and CoralMech. Updated getter methods to return typed values from TalonFXS methods directly, simplifying the code and improving type safety.

Ben-Vs-Bread committed:
working on implementing motion magic into elevator mech

Commit:
Google Java Format

notcharlie2026 committed:
Refactor ArmMech to use unit-safe types

Updated ArmMech subsystem methods to return and operate on unit-safe types (Angle, AngularVelocity, Voltage, Current, Temperature) instead of raw doubles. This improves type safety and clarity when handling physical quantities throughout the subsystem.

notcharlie2026 committed:
Refactor Tale subsystems for consistent naming and CAN bus

Updated AlgaeMech and CoralMech to use consistent member variable naming (m_ prefix) and centralized CAN bus name. ArmMech and ElevatorMech now use a static CAN bus name variable and remove unused status signals, simplifying periodic methods. Added Motion Magic control to ArmMech, and updated ElevatorMech to return unit-wrapped values for position, velocity, voltage, current, and temperature.

Ben-Vs-Bread committed:
in the middle

notcharlie2026 committed:
Refactor ArmMech and ElevatorMech to use Motion Magic

Removed direct position, velocity, and voltage control methods from ArmMech and ElevatorMech, consolidating movement logic to use Motion Magic requests. Updated command methods to use Motion Magic for target setting, simplified command creation, and improved calculation of positions and heights. This refactor streamlines control logic and prepares both subsystems for more consistent and maintainable motion profiling.

notcharlie2026 committed:
Update CoralMech.java

notcharlie2026 committed:
Refactor feedback sensor config for Arm and Elevator Note: elevator Cancoder and Armcancoder need to have a sensor to mech ratio thing

Updated ArmMech and ElevatorMech to configure TalonFX controllers to use fused CANcoder as the feedback sensor. Refactored MotionMagicExpoVoltage usage to instantiate requests locally instead of as class fields. Cleaned up CANcoder configuration and imports, and removed unused variables.

notcharlie2026 committed:
did the m_ thing 😎

Ben-Vs-Bread committed:
simulating motion magic for elevator

witcpalek commented:
just putting it in now without resolving all of the issues so we can create a base system core repo
